packages/plugin-core/src/test/suite-integ/MoveNoteCommand.test.ts

```diff
@@ -516,41 +530,174 @@ suite("MoveNoteCommand", function () {
             path.join("vault2", "foo.md")
           )
         ).toBeTruthy();
+
         // note not in old vault
+        await vaultAsserter(vault1).noMatch(["foo.md"]);
+        vaultNoteAsserter(vault1).noMatchByFName("foo");
+
+        // In new vault
+        vaultNoteAsserter(vault2).matchByFName("foo");
+        await vaultAsserter(vault2).match(["foo.md"]);
+
+        done();
+      },
+    });
+  });
+
+  test("bulk-move: move 2 notes from different vaults to new vault", (done) => {
+    runLegacyMultiWorkspaceTest({
+      ctx,
+      preSetupHook: async ({ wsRoot, vaults }) => {
+        await ENGINE_HOOKS_MULTI.setupBasicMulti({ wsRoot, vaults });
+      },
+      onInit: async ({ engine, vaults, wsRoot }) => {
+        function vaultAsserter(vault: DVault) {
+          return new VaultAsserter({ wsRoot, vault });
+        }
+        function vaultNoteAsserter(vault: DVault) {
+          return new NoteAsserter({
+            workspaceVault: { wsRoot, vault },
+            notes: engine.notes,
+          });
+        }
+
+        const notes = engine.notes;
+        const vault1 = vaults[0];
+        const vault2 = vaults[1];
+        const vault3 = vaults[2];
+
+        const fooNote = NoteUtils.getNoteByFnameV5({
+          fname: "foo",
+          notes,
+          vault: vault1,
+          wsRoot,
+        }) as NoteProps;
+
+        await VSCodeUtils.openNote(fooNote);
+        const cmd = new MoveNoteCommand();
+
+        // @ts-ignore
+        VSCodeUtils.showQuickPick = async () => {
+          return "proceed";
+        };
```

Comment:
this could be mocked with `sinon`.

`sinon.stub(VSCodeUtils, "showQuickPick").resolves("proceed");`

Reply:
done.